### PR TITLE
feat[Spanner]: Add samples for new features

### PR DESCRIPTION
**Pull request by amanda-tarafa**
Tests won't pass until https://github.com/googleapis/google-cloud-dotnet/pull/12164 is merged and release. I'm currently running them locally.

### CODE REVIEW

**Review by jskeet — DISMISSED** [review]
Nice and simple :)

**Comment by amanda-tarafa** [comment]
@jskeet I've re-requested review as I fixed a copyright year and the previous approval was lost.

**Review by jskeet — APPROVED** [review]

**Comment by amanda-tarafa** [comment]
The first commit here is from #2588 and will be rebased away once that's merged. We need it in the meantime so these tests run.

**Comment by amanda-tarafa** [comment]
@jskeet Here are the dependency updates for Spanner samples and the samples themselves, you had already reviewed one sample but not the other.
And also, when updating the dependencies I had to fix a test. PTAL, thanks!

File: spanner/api/Spanner.Samples/CommitDelayAsync.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                    { "AlbumId", SpannerDbType.Int64, 2 },
+                    { "AlbumTitle", SpannerDbType.String, "Go, Go, Go" }
+                });
+            insertAlbumCmd.Transaction = transaction;
+            rowsInserted += await insertAlbumCmd.ExecuteNonQueryAsync();
+
+            return rowsInserted;
+        });
+
+        Console.WriteLine("Data inserted.");
```
Comment (jskeet) [inline spanner/api/Spanner.Samples/CommitDelayAsync.cs:57]: I'm not sure how this is compiling - isn't this unreachable code? (We've got a return statement just above it.)

**Review by jskeet — APPROVED** [review]

File: spanner/api/Spanner.Samples/CommitDelayAsync.cs (hunk 1 above)
Comment (amanda-tarafa) [reply spanner/api/Spanner.Samples/CommitDelayAsync.cs:57]: Well it's compiling also in local. But you are right it is unreachable, I'll remove it (and I'll check from where I copied it.)

**Review by jskeet — APPROVED** [review]

**Comment by amanda-tarafa** [comment]
Failures are unrelated and being tracked in #1623 . Force merging.

**Pull request merged**